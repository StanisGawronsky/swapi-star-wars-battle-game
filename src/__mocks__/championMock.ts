export const championMock = {
  name: "Taun We",
  height: 213,
  mass: 2,
  hair_color: "none",
  skin_color: "grey",
  eye_color: "black",
  birth_year: 2,
  gender: "female",
};
