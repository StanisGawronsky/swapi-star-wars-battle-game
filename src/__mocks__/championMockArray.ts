import { ChampionType } from "~/types/ChampionType";

export const championMockArray: ChampionType[] = [
  {
    name: "Taun We",
    height: 213,
    mass: 2,
    hair_color: "none",
    skin_color: "grey",
    eye_color: "black",
    birth_year: 2,
    gender: "female",
  },
  {
    name: "VÃ³lk",
    height: 321,
    mass: 2,
    hair_color: "black",
    skin_color: "white",
    eye_color: "brown",
    birth_year: 92,
    gender: "male",
  },
];
