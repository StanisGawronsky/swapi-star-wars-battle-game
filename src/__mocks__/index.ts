export * from "./championMock";
export * from "./starshipMock";
