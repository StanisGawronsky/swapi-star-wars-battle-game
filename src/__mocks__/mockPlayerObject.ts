import { PlayerObject } from "~/types/PlayerTypes";

export const playerStarship: PlayerObject = {
  playerObject: {
    name: "Darth Vader",
    height: 202,
    mass: 136,
    hair_color: "none",
    skin_color: "white",
    eye_color: "yellow",
    birth_year: 41.9,
    gender: "male",
  },
  playerCard: {
    name: "Darth Vader",
    height: "202",
    mass: "136",
    hair_color: "none",
    skin_color: "white",
    eye_color: "yellow",
    birth_year: "41.9BBY",
    gender: "male",
  },
};
