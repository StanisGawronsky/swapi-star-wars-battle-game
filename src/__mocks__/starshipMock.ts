export const starshipMock = {
  name: "Executor",
  model: "Executor-class star dreadnought",
  manufacturer: "Kuat Drive Yards, Fondor Shipyards",
  cost_in_credits: 1143350000,
  length: 19000,
  max_atmosphering_speed: 0,
  crew: 279144,
  passengers: 38000,
  cargo_capacity: 250000000,
  consumables: 6,
  hyperdrive_rating: 2,
  MGLT: 40,
};
