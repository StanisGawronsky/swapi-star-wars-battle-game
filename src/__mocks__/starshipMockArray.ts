import { StarshipType } from "../types";

export const starshipMockArray: StarshipType[] = [
  {
    name: "Executor",
    model: "Executor-class star dreadnought",
    manufacturer: "Kuat Drive Yards, Fondor Shipyards",
    cost_in_credits: 1143350000,
    length: 19000,
    max_atmosphering_speed: 0,
    crew: 279144,
    passengers: 38000,
    cargo_capacity: 250000000,
    consumables: 6,
    hyperdrive_rating: 2,
    MGLT: 40,
  },
  {
    name: "Sentinel-class landing craft",
    model: "Sentinel-class landing craft",
    manufacturer: "Sienar Fleet Systems, Cyngus Spaceworks",
    cost_in_credits: 240000,
    length: 38,
    max_atmosphering_speed: 1000,
    crew: 5,
    passengers: 75,
    cargo_capacity: 180000,
    consumables: 30,
    hyperdrive_rating: 1,
    MGLT: 70,
  },
];
