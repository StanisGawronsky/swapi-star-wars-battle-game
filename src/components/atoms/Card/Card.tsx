import React, { useState } from "react";
import { playerTypes } from "~/types/Unions";
import { nanoid } from "nanoid";
import styles from "./styles/styles.module.scss";
import { parseAttributesForView } from "~/helpers/parseAttributesForView";
import clsx from "clsx";
interface IProps {
  player: playerTypes;
  currentAttribute: string;
}

const Card: React.FC<IProps> = ({ player, currentAttribute }: IProps) => {
  const list: Array<JSX.Element> = [];
  if (player) {
    for (const [key, value] of Object.entries(player)) {
      list.push(
        <p className={styles.attributeRow} key={nanoid()}>
          <span className={styles.attribute}>
            {parseAttributesForView(key)} :
          </span>
          <span className={styles.value}>{value}</span>
          <span
            className={clsx(
              currentAttribute === key && styles.currentAttribute
            )}
          ></span>
        </p>
      );
    }
  }
  return <div className={styles.cardWrapper}>{list}</div>;
};

export default Card;
