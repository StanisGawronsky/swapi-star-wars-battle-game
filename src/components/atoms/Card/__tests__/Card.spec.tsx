import React from "react";
import { render } from "@testing-library/react";
import Card from "../Card";
import { championMock, starshipMock } from "~/__mocks__/index";

describe("<Card />", () => {
  it("renders correctly Champion Card", () => {
    const { container } = render(<Card player={championMock} />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly Starship Card", () => {
    const { container } = render(<Card player={starshipMock} />);
    expect(container).toMatchSnapshot();
  });
});
