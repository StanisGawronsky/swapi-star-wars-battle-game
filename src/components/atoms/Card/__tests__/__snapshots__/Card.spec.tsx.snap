// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly Champion Card 1`] = `
<div>
  <div>
    <p>
      <span>
        name
         :
      </span>
      <span>
        Taun We
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        height
         :
      </span>
      <span>
        213
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        mass
         :
      </span>
      <span>
        2
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        hair color
         :
      </span>
      <span>
        none
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        skin color
         :
      </span>
      <span>
        grey
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        eye color
         :
      </span>
      <span>
        black
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        birth year
         :
      </span>
      <span>
        2
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        gender
         :
      </span>
      <span>
        female
      </span>
      <span
        class=""
      />
    </p>
  </div>
</div>
`;

exports[`<Card /> renders correctly Starship Card 1`] = `
<div>
  <div>
    <p>
      <span>
        name
         :
      </span>
      <span>
        Executor
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        model
         :
      </span>
      <span>
        Executor-class star dreadnought
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        manufacturer
         :
      </span>
      <span>
        Kuat Drive Yards, Fondor Shipyards
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        cost in credits
         :
      </span>
      <span>
        1143350000
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        length
         :
      </span>
      <span>
        19000
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        max atmosphering speed
         :
      </span>
      <span>
        0
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        crew
         :
      </span>
      <span>
        279144
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        passengers
         :
      </span>
      <span>
        38000
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        cargo capacity
         :
      </span>
      <span>
        250000000
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        consumables
         :
      </span>
      <span>
        6
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        hyperdrive rating
         :
      </span>
      <span>
        2
      </span>
      <span
        class=""
      />
    </p>
    <p>
      <span>
        MGLT
         :
      </span>
      <span>
        40
      </span>
      <span
        class=""
      />
    </p>
  </div>
</div>
`;
